File: curso-python/ex058.py
# Melhore o jogo do desafio 028 onde o computador vai "pensar" em um número entre 0 e 10
# só que agora o jogador vai tentar adivinhar até acertar, mostrando no final quantos palpites foram necesários para vencer
import random
from time import sleep

computador = random.randint(0,40)

print('='*20)
print("JOGO DA ADIVINHAÇÃO")
print("="*20)

print("olá jogador! vou pensar em um número entre 0 e 40, tente adivinhar...")

print('='*20)
print("pensando...")
print("="*20)
sleep(2)


tot_palpite = 1
palpite_jogador = int(input("qual seu palpite: "))

while palpite_jogador != computador:
    if palpite_jogador > computador:
        print("Menos... Tente mais uma vez")
    elif palpite_jogador < computador:
        print("Mais... tente mais uma vez")
    tot_palpite += 1
    palpite_jogador = int(input("qual seu palpite: "))

    
print(f"Acertou com {tot_palpite} tentativas. parabéns!")

File: javascript-exercicios/ex058.js
/*
# Melhore o jogo do desafio 028 onde o computador vai "pensar" em um número entre 0 e 10
# só que agora o jogador vai tentar adivinhar até acertar, mostrando no final quantos palpites foram necesários para vencer
*/

const prompt = require("prompt-sync")();
let computador = Math.floor(Math.random()*41);

console.log("sou seu computador... pensei em umm número entre 0 e 40");
console.log("será que você consegue adivinhar qual foi?");

let acertou = false;
let palpites = 0;

while (!acertou){
    let jogador = Number(prompt("qual é seu palpite?: "));
    palpites += 1;
    if (jogador == computador){
        acertou = true
    }else{
        if(jogador < computador){
            console.log("mais... tente mais uma vez")
        }else if (jogador > computador){
            console.log("menos... tente mais uma vez")
        }
    }
}

console.log(`acertou com ${palpites} tentativas.`)